src/nbl/ui/CWindowX11.cpp

```diff
@@ -227,6 +230,84 @@ CWindowX11::CWindowX11(core::smart_refctd_ptr<system::ISystem>&& sys, uint32_t _
 
     m_dpy = dpy;
     m_native = win;
+
+    XContext classId = XUniqueContext();
+    XSaveContext(m_dpy, m_native, classId, (XPointer)this);
```

Comment:
use @Crisspl's dynamic `x11` function caller for any `X` functions (dont want to compile with that lib being linked at compile time)

Reply:
yep, he knows, @sadiuk add more functions to the caller in case they're missing (not all x11 fucntions are being loaded)